libvast/src/system/read_query.cpp

```diff
@@ -70,17 +71,12 @@ read_query(const invocation& inv, std::string_view file_option,
       std::cerr << "please enter a query and confirm with CTRL-D: "
                 << std::flush;
     assign_query(std::cin);
+  } else if (inv.arguments.size() == argument_offset + 1) {
+    result = inv.arguments[argument_offset];
   } else {
-    // Assemble expression from all remaining arguments.
-    if (inv.arguments.size() > 1) {
-      VAST_WARN("spreading a query over multiple arguments is "
-                "deprecated; please pass it as a single string "
-                "instead.");
-      VAST_VERBOSE("(hint: use a heredoc if you run into quoting "
-                   "issues.)");
-    }
-    result = detail::join(inv.arguments.begin() + argument_offset,
-                          inv.arguments.end(), " ");
+    VAST_ERROR("spreading a query over multiple arguments is "
+               "not allowed; please pass it as a single string "
+               "instead.");
```

Comment:
We should observe this closely before the release. I've been hit a few times with the deprecation warning even though I passed just a single string.

Reply:
> fixes read_query for single argument queries for the pivot command.

See here. That was fixed by the argument_offset change in the diff.